plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt

```diff
@@ -287,25 +297,32 @@ class CodeWhispererCodeScanSession(val sessionContext: CodeScanSessionContext) {
 
     @Throws(IOException::class)
     fun uploadArtifactToS3(url: String, uploadId: String, fileToUpload: File, md5: String, kmsArn: String?, requestHeaders: Map<String, String>?) {
-        val uploadIdJson = """{"uploadId":"$uploadId"}"""
-        HttpRequests.put(url, "application/zip").userAgent(AwsClientManager.getUserAgent()).tuner {
-            if (requestHeaders.isNullOrEmpty()) {
-                it.setRequestProperty(CONTENT_MD5, md5)
-                it.setRequestProperty(CONTENT_TYPE, APPLICATION_ZIP)
-                it.setRequestProperty(SERVER_SIDE_ENCRYPTION, AWS_KMS)
-                if (kmsArn?.isNotEmpty() == true) {
-                    it.setRequestProperty(SERVER_SIDE_ENCRYPTION_AWS_KMS_KEY_ID, kmsArn)
-                }
-                it.setRequestProperty(SERVER_SIDE_ENCRYPTION_CONTEXT, Base64.getEncoder().encodeToString(uploadIdJson.toByteArray()))
-            } else {
-                requestHeaders.forEach { entry ->
-                    it.setRequestProperty(entry.key, entry.value)
+        try {
+            val uploadIdJson = """{"uploadId":"$uploadId"}"""
+            HttpRequests.put(url, "application/zip").userAgent(AwsClientManager.getUserAgent()).tuner {
+                if (requestHeaders.isNullOrEmpty()) {
+                    it.setRequestProperty(CONTENT_MD5, md5)
+                    it.setRequestProperty(CONTENT_TYPE, APPLICATION_ZIP)
+                    it.setRequestProperty(SERVER_SIDE_ENCRYPTION, AWS_KMS)
+                    if (kmsArn?.isNotEmpty() == true) {
+                        it.setRequestProperty(SERVER_SIDE_ENCRYPTION_AWS_KMS_KEY_ID, kmsArn)
+                    }
+                    it.setRequestProperty(SERVER_SIDE_ENCRYPTION_CONTEXT, Base64.getEncoder().encodeToString(uploadIdJson.toByteArray()))
+                } else {
+                    requestHeaders.forEach { entry ->
+                        it.setRequestProperty(entry.key, entry.value)
+                    }
                 }
+            }.connect {
+                val connection = it.connection as HttpURLConnection
+                connection.setFixedLengthStreamingMode(fileToUpload.length())
+                IoUtils.copy(fileToUpload.inputStream(), connection.outputStream)
             }
-        }.connect {
-            val connection = it.connection as HttpURLConnection
-            connection.setFixedLengthStreamingMode(fileToUpload.length())
-            IoUtils.copy(fileToUpload.inputStream(), connection.outputStream)
+        } catch (e: IOException) {
```

Comment:
Why do we catch and rethrow here?

Reply:
Removed try catch at parent function `createUploadUrlAndUpload` and tried to add this at individual step to get precise error message to telemetry.

Reply:
Do we need the catch blocks here?

Reply:
We need catch block to throw uploadArtifactFailedError but we can removed catch for IOException.

Reply:
But There is a potential chance of IOException so I handled IOException too in the catch block.